atlas-core/src/main/java/io/atlasmap/core/BaseAtlasModule.java

```diff
@@ -345,12 +351,11 @@ protected void processCombineField(AtlasSession session, Mapping mapping, List<F
                 addAudit(session, outputField.getDocId(), String
                         .format("Combine requires String field type for inputField.path=%s", inputField.getPath()),
                         outputField.getPath(), AuditStatus.WARN, null);
-                continue;
-            }
 
-            if (combineValues == null) {
-                // We need to support a sorted map w/ null values
-                combineValues = new HashMap<Integer, String>();
+                if (getConversionService().isPrimitive(inputField.getFieldType()))
+                    combineValues.put(inputField.getIndex(), inputField.getValue().toString());
```

Comment:
Let's take even more convenient path, pick up a converter if available, which supports some of non-primitive types. Here is the example
https://github.com/atlasmap/atlasmap-runtime/blob/d4e7b4b69964f7285ccd66a4c1b14c61f3675982/atlas-core/src/main/java/io/atlasmap/core/BaseAtlasModule.java#L402-L413

Reply:
You may want to use Mockito instead. Here is the example
https://github.com/atlasmap/atlasmap-runtime/blob/d4e7b4b69964f7285ccd66a4c1b14c61f3675982/atlas-itests-parent/atlas-itests-core/src/test/java/io/atlasmap/core/issue/CamelAtlasmap14Test.java#L45-L53

Reply:
You mean instead of implementation dummy classes ? (e.g https://github.com/mmelko/atlasmap-runtime/blob/combine_tweaks/atlas-core/src/test/java/io/atlasmap/core/CombineFromAtlasModuleTest.java#L78) ?

Reply:
Right, ah, the link to the specific line disappeared after your new commit :)